opencensus.go

```diff
@@ -85,6 +86,58 @@ func (c composableRegister) Register(ctx context.Context, cfg Config, vs []*view
 	c.setDefaultSampler(cfg.SampleRate)
 	c.setReportingPeriod(time.Duration(cfg.ReportingPeriod) * time.Second)
 
+	// modify metric tags
+	// ref: https://godoc.org/go.opencensus.io/plugin/ochttp#pkg-variables
+	if cfg.Exporters.Prometheus != nil {
+		for _, view := range vs {
+			// client metrics (method + statuscode tags are enabled by default)
+			if strings.Contains(view.Name, "http/client") {
+				// Host
+				if cfg.Exporters.Prometheus.HostTag {
+					view.TagKeys = appendIfMissing(view.TagKeys, ochttp.KeyClientHost)
+				}
+
+				// Path
+				if cfg.Exporters.Prometheus.PathTag {
+					view.TagKeys = appendIfMissing(view.TagKeys, ochttp.KeyClientPath)
+				}
+
+				// Method
+				if cfg.Exporters.Prometheus.MethodTag {
+					view.TagKeys = appendIfMissing(view.TagKeys, ochttp.KeyClientMethod)
+				}
+
+				// StatusCode
+				if cfg.Exporters.Prometheus.StatusCodeTag {
+					view.TagKeys = appendIfMissing(view.TagKeys, ochttp.KeyClientPath)
+				}
+			}
+
+			// server metrics
+			if strings.Contains(view.Name, "http/server") {
+				// Host
+				if cfg.Exporters.Prometheus.HostTag {
+					view.TagKeys = appendIfMissing(view.TagKeys, ochttp.Host)
+				}
+
+				// Path
+				if cfg.Exporters.Prometheus.PathTag {
+					view.TagKeys = appendIfMissing(view.TagKeys, ochttp.Path)
```

Comment:
same problem with the cardinality

Reply:
For client metrics - `ochttp.Transport` has a hardcoded call to  statsTransport in https://github.com/census-instrumentation/opencensus-go/blob/master/plugin/ochttp/client.go#L98

And that calls the following method which overwrites all tags and tracks them, we do not have any chance to modify them here:
https://github.com/census-instrumentation/opencensus-go/blob/master/plugin/ochttp/client_stats.go#L35

So i will have to copy a lot of ochttp over to modify that part.